### ISSUE TITLE
Blog post

### ISSUE THREAD

**Issue by chmac** [Trustroots/nostroots#21]
@guaka started writing a blog post...

tl;dr - We want to make sure that Trustroots exists 2 decades from now and unlock sharing beyond hospitality: rides, bicycles, meals, parties, etc.

Trustroots has already had a brush with death. The project lead stepped away without a word. Only one other person had access to most of the servers. We've seen the same issue with other hospitality sites. CouchSurfing took venture capital investment which slowly rotted out the site. Warm showers had major management clashes. Hospitality Club has had long periods of downtime as there's only one person running it.

Instead of just having one Trustroots, with one set of administrators, we want to have many. A network of connected sites. Maybe focused on different things. Some focused on bike tourers. Some on sharing rides. But each user's data, it lives across multiple sites. The user themselves has control over the data. If one site dies, you can log in somewhere else and still keep all of your connections, messages, and references. No one person can step away from the project and have it all come crashing down.

Plus, people can start sites with new features. Instead of sites like couchers starting over and building a network from scratch, they could start from the Trustroots users. When trustroots user logs into to a new site, all of their data comes with them. This is the major challenge for all new networks. When there's nobody on the network, it's not very useful. This makes it hard to experiment, to try new features. Hopefully, we can, slowly, make this easier.

We hope this is possible on something called nostr. We've looked at other decentralised technologies, but nostr is the first option that seems like it might work.

We don't have the illusion that we can move all of Trustroots over to nostr in a year. But we don't need to. We can enable small parts and then others can build on these foundations.

**Comment by guaka** [comment]
OK I like your version more!

I'll drop my version here and continue with yours :)

----------

In 2005 I came across an abandoned wiki I saved it from spammers and [which later became](https://hitchwiki.org/en/Hitchwiki:About#Background) Hitchwiki. This was around the 9 months or so that I was working on couchsurfing(tm), back when it wasn't (tm)ed yet, and I already wanted to do [a lot more](https://ideas.trustroots.org/2014/12/27/taking-roots/) than just enabling people to share couches. This didn't change when we launched Trustroots, kickstarting it from Hitchwiki and the community around it.  But there was no clear path towards enabling a wider, adventurous gift economy. An open API could have been nice but it would still have been quite centralized. About 2 years ago another danger of centralization appeared. After the main person left the project without any communication I was the only person left with access to the server and thus the database.

I made sure to share the keys to the castle with reliable long-time friends, but this is not enough to make sure Trustroots will be around 2 decades from now. So we started looking for ways to adding robustness to the project by (slow) decentralization. We looked at secure scuttlebutt, but anything built on that is barely even usable by geeks. We looked at the ActivityPub and Matrix protocols. But in the end both of these are federated and thus still rely on a person (or group of people) running one or more servers, which kinda boils down to the problem that we encountered which we wanted to solve. Gradually we want to take away the need to log into someone's server with a username and password.

Enter Nostr, "a decentralized network protocol for a distributed social networking system. The name is an acronym for Notes and Other Stuff Transmitted by Relays".  It's a protocol that gives control of identity and data back to the user and it doesn't rely heavily on domain names or specific servers.   Now we don't have the illusion that we can completely migrate all that is Trustroots now onto Nostr in one year. But we don't need to. We can enable small parts of Nostr, which others can then build upon.

Concrete ideas

Through serendipity I met the person who started and who is still running [bikesurf.org](https://bikesurf.org/), a fantastic bike sharing project that runs in Berlin and a couple of other cities. Currently it is mandatory to upload a scan of your ID, and no one really likes this solution. We have been talking how we can enable people to "take their Trustroots trust" over into Bikesurf ([thru NIP-05](https://github.com/Trustroots/nostroots/issues/4) and some [tech that we have to probably come up with ourselves](https://github.com/Trustroots/nostroots/issues/20)) instead of the ID verification.

[Kenny](https://www.hobolifestyle.com/) has been living out of a backpack since 2007. He's also created various cool projects around traveling and more specifically hitchhiking. He's already hacked a little around the (not so open) Trustroots API to enable location sharing from an Android app. He's coming to the [collective](https://collective.trustroots.org/) next month and I can imagine we can work out location sharing for folks on Trustroots in a way that can scale up.  This can be great for hitchhiking, and for people who enjoy picking up hitchhikers. It can also create a more serendipitous form of hospitality exchange. For example, I'd love to get a notification if someone who is in my Trustroots circles is say within a 100km radius so I can invite them to come over for a meal or stay a couple of nights.

Other possibilities

There are already 100s of projects on GitHub, including [some clients](https://nostr.net/#clients) that can be an interesting starting point to quickly experiment with ideas, e.g. a forum thru [satellite.earth](https://satellite.earth/), chat thru [nostrchat.io](https://www.nostrchat.io/) or [0xchat-app](https://github.com/0xchat-app). By adding geo data to notes, and turning circles into simple tags we can open up a lot more possibilities that anyone will be able to build upon.

**Comment by guaka** [comment]
published https://ideas.trustroots.org/2024/04/10/moving-trustroots-onto-nostr/
